src/wallet/wallet.cpp

```diff
@@ -3356,7 +3356,7 @@ bool CWallet::CreateCollateralTransaction(CMutableTransaction& txCollateral, std
     return true;
 }
 
-bool CWallet::GetBudgetSystemCollateralTX(CWalletTx& tx, uint256 hash, CAmount amount, bool fUseInstantSend)
+bool CWallet::GetBudgetSystemCollateralTX(CWalletTx& tx, uint256 hash, CAmount amount, bool fUseInstantSend, COutPoint outpoint)
```

Comment:
Could pass a reference here
```suggestion
bool CWallet::GetBudgetSystemCollateralTX(CWalletTx& tx, uint256 hash, CAmount amount, bool fUseInstantSend, const COutPoint& outpoint)
```

Reply:
Is the const a necessary change? what benefit is there to pass a reference in this case?

Reply:
In general, you should try to avoid creating a copy of a complex object when passing it to some function. It's also nice to specify that it can't be changed inside (hence `const`). If you are wondering why `hash` is not passed the same way  - it should too actually, we just missed this :)

Reply:
Also, `const MyClass& obj` allows passing rvalue references, .e.g `GetBudgetSystemCollateralTX(..., COutPoint())`, which is not allowed with non-const references.